### PR TITLE
SAAS-900 Put ToS under social login

### PR DESCRIPTION
**Pull request by TomasBlazek3PG**

### CODE REVIEW

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx [hunk 1]
```diff
@@ -74,10 +74,45 @@ interface ContextProps {
...
+  if (widgetRef.querySelector('.tos-label')) {
     return;
   }
 
+  if (controller === 'registration') {
+    signUpToS(widgetRef);
+  }
+
+  if (controller === 'idp-discovery') {
+    socialToS(widgetRef);
```
Comment (fabio-silva) [inline src/components/OktaSignInWidget/OktaSignInWidget.tsx:86]: Please declare those functions before calling them here, for two reasons:

- Both are function expressions, they are not hoisted;
- Both are `const`. Even if they were hoisted, they would have been unitialized and a reference error might be thrown.

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx [hunk 2]
```diff
@@ -74,10 +74,45 @@ interface ContextProps {
...
+  }
+
+  const tosWrapper = document.createElement('div');
+
+  container.append(tosWrapper);
+  ReactDOM.render(
+    // eslint-disable-next-line jsx-a11y/label-has-associated-control
+    <p className='tos-label'>
+      {Messages.byContinuing}
+      <a href={TERMS_OF_SERVICE_URL} target='_blank' rel='noreferrer'>
```
Comment (fabio-silva) [inline src/components/OktaSignInWidget/OktaSignInWidget.tsx:104]: For compatibility reasons, please add to both anchors: `rel='noreferrer noopener'`

File: src/pages/Login/Login.styles.ts [hunk 3]
```diff
@@ -117,6 +117,16 @@ export const getStyles = (theme: GrafanaTheme) => {
...
           }
         }
+
+        .social-auth-button {
+          padding-left: 50px;
+          padding-right: 0;
+          text-align: center;
+        }
+
+        .tos-label a, .tos-label a {
```
Comment (fabio-silva) [inline src/pages/Login/Login.styles.ts:127]: Duplicate?

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: src/pages/Login/Login.styles.ts (hunk 3 above)
Comment (TomasBlazek3PG) [reply src/pages/Login/Login.styles.ts]: Oh yeah it was suppose to be label a, p a but then I decided to use the class but forgot to get rid of the duplicate. Thanks.

**Review by fabio-silva — APPROVED** [review]

File: src/pages/Login/Login.styles.ts (hunk 3 above)
Comment (fabio-silva) [reply src/pages/Login/Login.styles.ts]: 👍

File: src/components/OktaSignInWidget/OktaSignInWidget.messages.ts [hunk 4]
```diff
@@ -1,5 +1,8 @@
 export const Messages = {
   iHaveAgreed: 'I have read and agree to the ',
+  byContinuing:
+    // eslint-disable-next-line max-len
```
Comment (nicolalamacchia) [inline src/components/OktaSignInWidget/OktaSignInWidget.messages.ts:4]: is this needed?

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx [hunk 5]
```diff
@@ -69,15 +69,34 @@ const ToSCheckbox: FC<ToSCheckboxProps> = ({ submitBtn }) => {
...
-  if (controller !== 'registration' || widgetRef.querySelector('.tos-label')) {
+  if (!container) {
     return;
   }
 
+  const tosWrapper = document.createElement('div');
+
+  container.append(tosWrapper);
+  ReactDOM.render(
+    // eslint-disable-next-line jsx-a11y/label-has-associated-control
```
Comment (nicolalamacchia) [inline src/components/OktaSignInWidget/OktaSignInWidget.tsx:83]: is this needed?

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx [hunk 6]
```diff
@@ -100,6 +119,24 @@ const insertToS = ({ controller }: ContextProps, widgetRef: HTMLDivElement) => {
...
+    return;
+  }
+
+  if (controller === 'registration') {
+    signUpToS(widgetRef);
+  }
+
+  if (controller === 'idp-discovery') {
+    socialToS(widgetRef);
+  }
```
Comment (nicolalamacchia) [inline src/components/OktaSignInWidget/OktaSignInWidget.tsx:137]: either use a `else if` here or return in the first if

**Review by nicolalamacchia — CHANGES_REQUESTED** [review]

**Review by nicolalamacchia — APPROVED** [review]

**Review by nicolalamacchia — APPROVED** [review]

**Pull request merged**